pkg/net/libp2p/libp2p_transport.go

```diff
@@ -158,6 +163,35 @@ func (t *Transport) Connect(ctx context.Context, nodes map[types.NodeID]types.No
 	t.connect(ctx, maputil.GetKeys(parsedAddrInfo))
 }
 
+// WaitFor polls the current connection state and returns when at least n connections have been established.
+func (t *Transport) WaitFor(n int) {
+	for {
+		t.connsLock.RLock()
+		numConnections := 0
+		for _, ci := range t.conns {
+			if ci.isConnected(t.host) {
+				numConnections++
+			}
+		}
+		t.connsLock.RUnlock()
+
+		// We subtract one, as we always assume to be connected to ourselves
+		// and the connection to self does not appear in t.conns.
+		if numConnections >= n-1 {
+			return
+		}
+		time.Sleep(connWaitPollInterval)
```

Comment:
polling ;-)

(This is just an ad-hoc implementation that we needed quickly, it can be done properly together with the next overhaul of the transport module that will need to happen at some point anyway.)